src/Cake.Core/CakeEngine.cs

```diff
@@ -232,16 +264,35 @@ private async Task RunTask(ICakeContext context, IExecutionStrategy strategy, Ca
         {
             stopWatch.Restart();
 
+            PublishEvent(BeforeSetup, new BeforeSetupEventArgs(context));
+#pragma warning disable 618
             PublishEvent(Setup, new SetupEventArgs(context));
+#pragma warning restore 618
 
-            if (_actions.Setups.Count > 0)
+            try
             {
-                foreach (var setup in _actions.Setups)
+                if (_actions.Setups.Count > 0)
                 {
-                    strategy.PerformSetup(setup, new SetupContext(context, targetTask, tasks));
-                }
+                    foreach (var setup in _actions.Setups)
+                    {
+                        PerformTaskSetup(context, strategy, targetTask, false);
+                        try
+                        {
+                            strategy.PerformSetup(setup, new SetupContext(context, targetTask, tasks));
+                        }
+                        catch (Exception e)
+                        {
+                            Console.WriteLine(e);
```

Comment:
This should not write to the Console. We should Log the exception to the logger - or remove the Try Catch entirely and let it bubble up

Reply:
Apologies if I'm missing something here, but my understanding is that the Setup runs only once at the start and before any TaskSetup, and TaskSetup runs on every task - Here it seems we're running TaskSetup multiple times and _before_ Setup, instead of after.

Reply:
I removed the try/catch, and removed the `PerformTaskSetup` call, so that logic is not changed